### PR TITLE
Document a few features for rel 1.10

### PR DESCRIPTION
**Pull request by RandySheriffH**
- Document cupti dependency introduced by cuda profiler;
- Document usage of cuda profiler;
- Document custom threading hooks

### CODE REVIEW

**Review by jingyanwangms — APPROVED** [review]

File: docs/performance/tune-performance.md [hunk 1]
```diff
@@ -51,6 +51,19 @@ In both cases, you will get a JSON file which contains the detailed performance
 * Type chrome://tracing in the address bar
 * Load the generated JSON file
 
+For Cuda EP, performance numbers from device will be attached to those from host. For example:
+```
+{"cat":"Node", "name":"Add_1234", "dur":17, ...}
+{"cat":"Kernel", "name":"ort_add_cuda_kernel", dur:33, ...}
+```
+Here, "Add" operator from host initiated a Cuda kernel on device named "ort_add_cuda_kernel" which lasted for 33 macroseconds.
```
Comment (faxu) [inline docs/performance/tune-performance.md:59]: What is macrosecond?

File: docs/performance/tune-performance.md [hunk 2]
```diff
@@ -51,6 +51,19 @@ In both cases, you will get a JSON file which contains the detailed performance
 * Type chrome://tracing in the address bar
 * Load the generated JSON file
 
+For Cuda EP, performance numbers from device will be attached to those from host. For example:
```
Comment (faxu) [inline docs/performance/tune-performance.md:54]: Please capitalize CUDA here and in other updated text as well

File: docs/performance/tune-performance.md [hunk 3]
```diff
@@ -148,6 +161,9 @@ Memory consumption can be reduced between multiple sessions by configuring the s
 * Inter op num threads (used only when parallel execution is enabled) is not affected by OpenMP settings and should
 always be set using the ORT APIs.
 
+### Custom threading hooks
+For customer prefers to create and manage their own threads, please see usage [here](https://github.com/microsoft/onnxruntime/blob/6284cbe833bc47eb90ed4082ff2ae015b8e5c004/onnxruntime/test/shared_lib/test_inference.cc#L1867) and [here](https://github.com/microsoft/onnxruntime/blob/6284cbe833bc47eb90ed4082ff2ae015b8e5c004/onnxruntime/test/global_thread_pools/test_main.cc#L36).
```
Comment (faxu) [inline docs/performance/tune-performance.md:165]: Instead of referring to line numbers that can change, can you directly include the example here?

Comment (faxu) [inline docs/performance/tune-performance.md:165]: Can reword this sentence more directly as "To create and manage custom threads..."

**Review by faxu — CHANGES_REQUESTED** [review]

File: docs/performance/tune-performance.md (hunk 1 above)
Comment (RandySheriffH) [reply docs/performance/tune-performance.md]: typo fixed.

File: docs/performance/tune-performance.md (hunk 3 above)
Comment (RandySheriffH) [reply docs/performance/tune-performance.md]: Examples attached.

File: docs/performance/tune-performance.md (hunk 2 above)
Comment (RandySheriffH) [reply docs/performance/tune-performance.md]: Done.

File: docs/performance/tune-performance.md (hunk 3 above)
Comment (RandySheriffH) [reply docs/performance/tune-performance.md]: How about "Occasionally, customers might prefer ..." ...

**Review by faxu — APPROVED** [review]

**Pull request merged**